sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/templating/JavaParserTemplateProcessor.java

```diff
@@ -307,14 +251,42 @@ private void configureInternalMethod(MethodDeclaration internalMethod, HttpReque
         BlockStmt body = internalMethod.getBody().get();
 
         initializeHttpRequest(body, method);
-        boolean serializationFormatSet = addRequestBody(body, method);
+        setContentType(body, method);
+        boolean serializationFormatSet = RequestBodyHandler.configureRequestBody(body, method.getBody(), processingEnv);
         addRequestOptionsToRequestIfPresent(body, method);
         finalizeHttpRequest(body, method.getMethodReturnType(), method, serializationFormatSet);
 
         internalMethod.setBody(body);
     }
 
-    // Helper methods
+    private void setContentType(BlockStmt body, HttpRequestContext method) {
+        final HttpRequestContext.Body requestBody = method.getBody();
+        if (requestBody == null || requestBody.getParameterType() == null) {
+            return;
+        }
+
+        boolean isContentTypeSetInHeaders
+            = method.getParameters().stream().anyMatch(p -> p.getName().equals("contentType"));
```

Comment:
Might be reading this wrong, but is this trying to find `@HeaderParam("Content-Type")`? If so, should the `getName()` check for `Content-Type`?

Also, flip the equality check to prevent possible NPEs.

```suggestion
            = method.getParameters().stream().anyMatch(p -> "contentType".equals(p.getName()));
```

Reply:
This is looking for header param, parameter defined in the method signature. Like so here https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/clientcore/annotation-processor-test/src/main/java/io/clientcore/annotation/processor/test/implementation/TestInterfaceClientImpl.java#L50-L51. I will switch the equals.

Reply:
Ah, given that `method.getParameters()` returns `List<MethodParameter>` we would be better doing a deep check on each method parameter.

```java
boolean isContentTypeSetInHeaders = method.getParameters().stream()
    .anyMatch(parameter -> {
        HeaderParam headerParam = parameter.getTypeMirror().getAnnotation(HeaderParam.class);
        if (headerParam != null) {
            return "Content-Type".equals(headerParam.value());
        }
                
        return false;
    });
```

This is inspecting that the parameter is annotated with `@HeaderParam` and that value (HTTP header name) is `Content-Type`. This is safer as its possible for a non-header parameter to have the name `contentType`, ex `void interfaceMethod(String contentType, @HeaderParam("Content-Type") httpContentType)`